### PR TITLE
Check for local draft when loading a reply box, if exists update a cloud draft with a local one

### LINKED ISSUE #3971: Local draft should override the cloud draft once user get back online
While working on #3126 and #3337 I faced this issue which might cause lost reply drafts in use-cases described below:

### Steps to reproduce

1. Open any secure email
2. Type "123" into the secure reply, ensure the cloud draft is saved normally (`Saved`)
3. Go offline
4. Type "456" into the same reply box, so the whole reply is now "123456", ensure the draft is saved locally (`Draft saved locally (offline)`)
5. Go to inbox
6. Go back online
7. Open the same secure email

### Expected behavior

The user sees the "123456" draft (local draft, if exists in the same thread, should override the cloud draft)

### Actual behavior

The user sees the "123" draft, the local draft is lost.

### PR DESCRIPTION
**Pull request by limonte**
This PR checks if a local draft exists when loading the reply box. If so, the local draft will override a cloud draft.

close #3971

----------------------------------

**Tests**:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/chrome/elements/compose-modules/compose-draft-module.ts [hunk 1]
```diff
@@ -141,6 +142,7 @@ export class ComposeDraftModule extends ViewModule<ComposeView> {
         } else {
           await this.doUploadDraftWithLocalStorageFallback(mimeMsg, msgData, async () => {
             await this.view.emailProvider.draftUpdate(this.view.draftId, mimeMsg, this.view.threadId);
+            await this.localDraftRemove(); // delete local draft if there is one
```
Comment (rrrooommmaaa) [inline extension/chrome/elements/compose-modules/compose-draft-module.ts:145]: have you considered moving this call to doUploadDraftWithLocalStorageFallback()?
Also, msgData parameter seems unused and can be refactored out

File: extension/chrome/elements/compose-modules/compose-draft-module.ts [hunk 2]
```diff
@@ -125,9 +128,7 @@ export class ComposeDraftModule extends ViewModule<ComposeView> {
         if (!this.view.draftId || this.isLocalDraftId(this.view.draftId)) {
           const draftId = await this.doUploadDraftWithLocalStorageFallback(mimeMsg, msgData, async () => {
             const { id } = await this.view.emailProvider.draftCreate(mimeMsg, this.view.threadId);
-            if (this.isLocalDraftId(this.view.draftId)) { // delete local draft if there is one
-              await storageLocalRemove([this.view.draftId]);
-            }
+            await this.localDraftRemove(); // delete local draft if there is one
```
Comment (rrrooommmaaa) [inline extension/chrome/elements/compose-modules/compose-draft-module.ts:131]: have you considered moving this call to doUploadDraftWithLocalStorageFallback()?

Comment (limonte) [reply extension/chrome/elements/compose-modules/compose-draft-module.ts]: Done 👍

File: extension/chrome/elements/compose-modules/compose-draft-module.ts (hunk 1 above)
Comment (limonte) [reply extension/chrome/elements/compose-modules/compose-draft-module.ts]: Done, thanks for noticing 👍

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**